tests/app/data_model/test_answer_store.py

```diff
@@ -0,0 +1,246 @@
+
+        self.assertEqual(len(self.store.answers), 3)
+
+    def test_updates_answer(self):
+        answer_1 = {
+            'block_id': "3",
+            'answer_id': "4",
+            'question_id': "5",
+            'answer_instance': 1,
+            'value': 25,
+        }
+        answer_2 = {
+            'block_id': "3",
+            'answer_id': "4",
+            'question_id': "5",
+            'answer_instance': 1,
+            'value': 65,
+        }
+
+        self.store.add(answer_1)
+        self.store.update(answer_2)
+
+        self.assertEqual(self.store.count(answer_2), 1)
+
+        store_match = self.store.filter(answer_1)
+
+        self.assertEqual(store_match, [answer_2])
+
+    def test_filters_answers(self):
+        answer_1 = {
+            'block_id': "1",
+            'answer_id': "2",
+            'question_id': "3",
+            'answer_instance': 1,
+            'value': 25,
+        }
+        answer_2 = {
+            'block_id': "1",
+            'answer_id': "5",
+            'question_id': "6",
+            'answer_instance': 1,
+            'value': 65,
+        }
+
+        self.store.add(answer_1)
+        self.store.add(answer_2)
+
+        filtered = self.store.filter({
+            'block_id': "1"
+        })
+
+        self.assertEqual(len(filtered), 2)
+
+        filtered = self.store.filter({
+            'question_id': "6"
+        })
+
+        self.assertEqual(len(filtered), 1)
+
+    def test_maps_answers(self):
+        answer_1 = {
+            'block_id': "1",
+            'answer_id': "2",
+            'question_id': "3",
+            'answer_instance': 1,
+            'value': 25,
+        }
+        answer_2 = {
+            'block_id': "1",
+            'answer_id': "5",
+            'question_id': "6",
+            'answer_instance': 1,
+            'value': 65,
+        }
+
+        self.store.add(answer_1)
+        self.store.add(answer_2)
+
+        expected_answers = {
+            "2_1": 25,
+            "5_1": 65
+        }
+
+        self.assertEqual(self.store.map(), expected_answers)
+
+    def test_maps_and_filters_answers(self):
+        answer_1 = {
+            'block_id': "1",
+            'answer_id': "2",
+            'question_id': "3",
+            'answer_instance': 1,
+            'value': 25,
+        }
+        answer_2 = {
+            'block_id': "1",
+            'answer_id': "5",
+            'question_id': "6",
+            'answer_instance': 1,
+            'value': 65,
+        }
+
+        self.store.add(answer_1)
+        self.store.add(answer_2)
+
+        expected_answers = {
+            "5_1": 65
+        }
+
+        self.assertEqual(self.store.map({
+            "question_id": "6"
+        }), expected_answers)
+
+    def test_returns_ordered_map(self):
+        answer = {
+            'block_id': "1",
+            'answer_id': "2",
+            'question_id': "3",
+            'value': 25,
+        }
+
+        for i in range(0, 1000):
```

Comment:
Belt and braces - a dict can potentially return in any order and may return the items in the order they were inserted, this test covers every eventuality. Suppose it could be reduced to 100, because even that will cover it.